pkg/base/helm.go

```diff
@@ -271,3 +274,105 @@ func checkChartForVersion(file *upstreamtypes.UpstreamFile) (string, error) {
 	// if no determination is made, assume v2
 	return "v2", nil
 }
+
+// insert namespace if it's defined in the spec and not already present in the manifests
+func kustomizeHelmNamespace(baseFiles []BaseFile, renderOptions *RenderOptions) ([]BaseFile, error) {
+	var updatedBaseFiles []BaseFile
+	if renderOptions.Namespace != "" {
+		chartsPath, err := ioutil.TempDir("", "charts")
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to create temp dir")
+		}
+		defer os.RemoveAll(chartsPath)
+
+		resources := map[string]BaseFile{}
+		var kustomizeResources []string
+		var kustomizePatches []kustomizetypes.PatchStrategicMerge
+		foundGVKNamesMap := map[string]bool{}
+		for _, baseFile := range baseFiles {
+			// write temp files for manifests that need a namespace
+			gvk, manifest := GetGVKWithNameAndNs(baseFile.Content, renderOptions.Namespace)
```

Comment:
looking at this function's definition, it seems like it would override the namespace of manifests with no namespace with the provided namespace here.

Reply:
i think that's intended behavior though. since we're adding that namespace to manifests with no namespace, it makes sense to me to add it to the gvk. if we didn't do that, splitting the manifests and replacing the `baseFile.Content` would fail after the kustomize step (since that second gvk would include the new namespace).